compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TileAttention.cpp

```diff
@@ -292,6 +304,103 @@ IREE::LinalgExt::AttentionOp tileAttention(IREE::LinalgExt::AttentionOp attnOp,
   return tiledAttentionOp;
 }
 
+void convertToOnlineAttention(IREE::LinalgExt::AttentionOp attnOp,
+                              SmallVectorImpl<Operation *> &ops,
+                              RewriterBase &rewriter) {
+  rewriter.setInsertionPoint(attnOp);
+
+  Location loc = attnOp.getLoc();
+  MLIRContext *ctx = attnOp.getContext();
+
+  FailureOr<AttentionOpDetail> maybeOpInfo =
+      AttentionOpDetail::get(attnOp.getIndexingMapsArray());
+  assert(succeeded(maybeOpInfo) && "Invalid attention indexing maps");
+  AttentionOpDetail opInfo = maybeOpInfo.value();
+
+  // Create standard maps for max and sum: (batch, m)
+  int64_t rank = opInfo.getDomainRank();
+  AffineMap maxMap = AffineMap::get(/*dimCount=*/rank, /*symbolCount=*/0, ctx);
+  for (auto dim :
+       llvm::concat<const int64_t>(opInfo.getBatchDims(), opInfo.getMDims())) {
+    maxMap = maxMap.insertResult(rewriter.getAffineDimExpr(dim),
+                                 maxMap.getNumResults());
+  }
+  AffineMap sumMap = maxMap;
+
+  SmallVector<Range> sizes = attnOp.getIterationDomain(rewriter);
+
+  // Create fill for acc, max and sum.
+  // TODO: Acc should not need a fill. The attention op should get a filled
+  // input instead of an empty input.
+  Value zeroAcc = rewriter.create<arith::ConstantOp>(
+      loc, rewriter.getZeroAttr(attnOp.getOutputType().getElementType()));
+  Value accFill =
+      rewriter
+          .create<linalg::FillOp>(loc, ValueRange{zeroAcc}, attnOp.getOutput())
+          .getResult(0);
```

Comment:
Can this be `rewriter.create<linalg::FillOp>(...).result()`? I think we have the method for `FillOpInterface`.

https://github.com/llvm/llvm-project/blob/7c8a8bdc3a555e531e79af6ea9a60eb30c41475d/mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td#L210-L213

Reply:
done!